src/core/border_router/routing_manager.cpp

```diff
@@ -624,20 +625,17 @@ const Ip6::Prefix *RoutingManager::EvaluateOnLinkPrefix(void)
     // the same smallest on-link prefix and the application-specific prefix is not used.
     else if (mIsAdvertisingLocalOnLinkPrefix)
     {
-        if (mLocalOnLinkPrefix < *smallestOnLinkPrefix)
-        {
-            newOnLinkPrefix = &mLocalOnLinkPrefix;
-        }
-        else
+        if (!(mLocalOnLinkPrefix < *smallestOnLinkPrefix))
         {
             LogInfo("EvaluateOnLinkPrefix: There is already smaller on-link prefix %s on interface %u",
                     smallestOnLinkPrefix->ToString().AsCString(), mInfraIfIndex);
             DeprecateOnLinkPrefix();
```

Comment:
Should we mIsAdvertisingLocalOnLinkPrefix = false; in DeprecateOnLinkPrefix ?

Reply:
Good idea. Will change in next push.